### PR TITLE
Ice.ServerIdleTime test fixes

### LINKED ISSUE #2025: Ice.ServerIdleTime in tests
A number of tests configure Ice.ServerIdleTime, however, it's not clear why they do it and no test appears to actually verify that Ice.ServerIdleTime works.

Keep in the mind the default value for Ice.ServerIdleTime is 0 - meaning, the communicator is not shutdown automatically when inactive.

### PR DESCRIPTION
**Pull request by externl**
This PR does the following `Ice.ServerIdleTime` updates:

- Remove the random usage in the tests
- Adds a test to exercise `Ice.ServerIdleTime`
- Fixes a bug affecting primarily `kqueue`  (we also need to fix this on 3.7 after)

Closes #2025

### CODE REVIEW

File: cpp/src/Ice/Selector.cpp [hunk 1]
```diff
@@ -639,7 +639,10 @@ Selector::select(int timeout)
     }
     else if (timeout > 0)
     {
+        // kpoll and select use seconds, epoll and poll use milliseconds
+#    if !defined(ICE_USE_KQUEUE) && !defined(ICE_USE_SELECT)
```
Comment (externl) [inline cpp/src/Ice/Selector.cpp:643]: When timespec is used we actually need seconds, no milliseconds. AFAICT this also affects 3.7.

File: cpp/test/Ice/faultTolerance/Server.cpp [hunk 2]
```diff
@@ -19,14 +19,7 @@ class Server : public Test::TestHelper
 void
 Server::run(int argc, char** argv)
 {
-    //
-    // In this test, we need a longer server idle time, otherwise
-    // our test servers may time out before they are used in the
-    // test.
-    //
     Ice::PropertiesPtr properties = createTestProperties(argc, argv);
```
Comment (bernardnormier) [inline cpp/test/Ice/faultTolerance/Server.cpp:22]: You can also remove the properties.

File: csharp/test/Ice/adapterDeactivation/AllTests.cs [hunk 3]
```diff
@@ -197,6 +197,25 @@ public static async Task<Test.TestIntfPrx> allTests(global::Test.TestHelper help
                     output.WriteLine("ok");
                 }
 
+                output.Write("testing idle server shutdown...");
```
Comment (bernardnormier) [inline csharp/test/Ice/adapterDeactivation/AllTests.cs:200]: See C++ comment

File: csharp/test/Ice/adapterDeactivation/AllTests.cs [hunk 4]
```diff
@@ -197,6 +197,25 @@ public static async Task<Test.TestIntfPrx> allTests(global::Test.TestHelper help
...
+                output.Flush();
+                {
+                    Ice.InitializationData initData = new Ice.InitializationData()
+                    {
+                        properties = communicator.getProperties().ice_clone_(),
+                    };
+                    initData.properties.setProperty("Ice.ServerIdleTime", "1");
+                    // The thread pool threads have to be idle first before server idle time is checked.
+                    initData.properties.setProperty("Ice.ThreadPool.Server.ThreadIdleTime", "1");
+                    using (Ice.Communicator idleCommunicator = Ice.Util.initialize(initData))
```
Comment (bernardnormier) [inline csharp/test/Ice/adapterDeactivation/AllTests.cs:210]: You should use a using statement here.

File: csharp/test/Ice/binding/Server.cs [hunk 5]
```diff
@@ -11,7 +11,6 @@ public class Server : TestHelper
             public override void run(string[] args)
             {
                 Ice.Properties properties = createTestProperties(ref args);
-                properties.setProperty("Ice.ServerIdleTime", "30");
```
Comment (bernardnormier) [inline csharp/test/Ice/binding/Server.cs:14]: Simplify further - no need to parse the args into properties.

File: csharp/test/Ice/enums/Server.cs [hunk 6]
```diff
@@ -11,7 +11,6 @@ public class Server : TestHelper
             public override void run(string[] args)
             {
                 Ice.Properties properties = createTestProperties(ref args);
-                properties.setProperty("Ice.ServerIdleTime", "30");
```
Comment (bernardnormier) [inline csharp/test/Ice/enums/Server.cs:14]: Same here

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**